File: base64decoder/python/base64_decode.py

import base64

encodedStr = "aGVsbG8gd29ybGQxMjMhPyQ="

# Standard Base64 Decoding
decodedBytes = base64.b64decode(encodedStr)
decodedStr = str(decodedBytes, "utf-8")
print(decodedStr)

# Url Safe Base64 Decoding
decodedBytes = base64.urlsafe_b64decode(encodedStr)
decodedStr = str(decodedBytes, "utf-8")
print(decodedStr)

File: base64decoder/golang/base64_decode.go
package main

import (
	"encoding/base64"
	"fmt"
)

func main() {
	data := "hello:world!?$*&()'-=@~"

	// Base64 Standard Decoding
	sEnc := base64.StdEncoding.EncodeToString([]byte(data))
	sDec, err := base64.StdEncoding.DecodeString(sEnc)
	if err != nil {
		fmt.Printf("Error decoding string: %s ", err.Error())
		return
	}
	fmt.Println(sDec)

	// Base64 Url Decoding
	uEnc := base64.URLEncoding.EncodeToString([]byte(data))
	uDec, err := base64.URLEncoding.DecodeString(uEnc)
	if err != nil {
		fmt.Printf("Error decoding string: %s ", err.Error())
		return
	}
	fmt.Println(uDec)

}
